src/plugins/share/public/components/context/index.tsx

```diff
@@ -31,6 +31,7 @@ export interface IShareContext extends ShareContext {
   isEmbedded: boolean;
   theme: ThemeServiceSetup;
   i18n: I18nStart;
+  disableServerlessSettings?: boolean;
```

Comment:
I suggest we use "positive-based" naming, as that tends to be more understandable. Also, to not refer to the constraint as one for serverless:
```suggestion
 publicApiEnabled?: boolean;
```

And then, in other places in this PR where the variable is referenced, you'll have to flip how the value is used, since it is currently "negation-based."